int32 ID1
int32 ID2